src/formats/hdr.rs

```diff
@@ -0,0 +1,61 @@
+use std::io::{self, BufRead, Seek, SeekFrom};
+
+use crate::{ImageResult, ImageSize};
+
+pub fn size<R: BufRead + Seek>(reader: &mut R) -> ImageResult<ImageSize> {
+    reader.seek(SeekFrom::Start(0))?;
+
+    // Read the first line and check if it's a valid HDR format identifier
+    let mut format_identifier = String::new();
+    reader.read_line(&mut format_identifier)?;
+    if !format_identifier.starts_with("#?RADIANCE") && !format_identifier.starts_with("#?RGBE") {
+        return Err(
+            io::Error::new(io::ErrorKind::InvalidData, "Invalid HDR format identifier").into(),
+        );
+    }
+
+    loop {
+        let mut line = String::new();
+        reader.read_line(&mut line)?;
+
+        // Extract width and height information
+        if line.trim().is_empty() || !line.starts_with("-Y") {
+            continue;
+        }
+
+        if line.starts_with("-Y") {
+            let dimensions: Vec<&str> = line.split_whitespace().collect();
+            if dimensions.len() != 4 {
+                return Err(io::Error::new(
+                    io::ErrorKind::InvalidData,
+                    "Invalid HDR dimensions line",
+                )
+                .into());
+            }
+
+            let height = dimensions[1].parse::<usize>().ok();
+            let width = dimensions[3].parse::<usize>().ok();
+
+            match width.is_some() && height.is_some() {
```

Comment:
Does it make more sense to turn this into

```rust
if width.is_some() && height.is_some() {
    return Ok(ImageSize { 
        width: width.unwrap(),
        height: height.unwrap(),
    });
}
```

Reply:
that's what I do at first, cargo clip will suggest a warning, so I changed to match...